c2corg_api/views/document_version.py

```diff
@@ -66,23 +69,28 @@ def _load_version(
         if version is None:
             raise HTTPNotFound('invalid version')
 
-        archive_document = version.document_archive
-        archive_document.geometry = version.document_geometry_archive
-        archive_document.locales = [version.document_locales_archive]
+        document = None
+        if not version.masked or self.request.has_permission('moderator'):
+            archive_document = version.document_archive
+            archive_document.geometry = version.document_geometry_archive
+            archive_document.locales = [version.document_locales_archive]
 
-        if adapt_schema:
-            schema = adapt_schema(schema, archive_document)
+            if adapt_schema:
+                schema = adapt_schema(schema, archive_document)
+
+            document = to_json_dict(
+                archive_document,
+                schema,
+                cook_locale=True
+            )
+        # TODO if masked, return limited info as in DocumentInfoRest?
```

Comment:
this TODO is not yet resolved?

Reply:
No, it's not: as for now when a version is masked, the API returns nothing as the archived documents. See https://github.com/c2corg/v6_api/pull/1393/files#diff-33ce4bfbec662d496fcd00822bfdb6174d23586982306f66c15194dba40f771cR72
Perhaps returning at least the version title as in `DocumentInfoRest` would be nice? See https://github.com/c2corg/v6_api/blob/969bf13ed8e4ad5f0e08694694394e0edba3f313/c2corg_api/views/document_info.py#L17-L20
I have not tried yet to figure out if this is easily feasible though.